packages/@aws-cdk/aws-iam/lib/principals.ts

```diff
@@ -48,6 +48,17 @@ export interface IPrincipal extends IGrantable {
    * question does not have a policy document to add the statement to.
    */
   addToPolicy(statement: PolicyStatement): boolean;
+
+  /**
+   * Return whether the principal and the given construct are in the same AWS account.
+   *
+   * This is used by `Grant` to determine whether a trust policy needs
+   * to be added to the resource as well as a permissions policy needs to be
+   * added to the identity.
+   *
+   * Returns `undefined` if it is unknown whether the accounts are the same.
+   */
+  sameAccount(scope: cdk.IConstruct): boolean | undefined;
```

Comment:
1. I feel that this API should accept a `cdk.Stack` and not a `cdk.IConstruct`. It will make this API much clearer.
2. Would it make sense to make this `_internal`? It's used only inside this module and TBH it feels like it's polluting the interface.